src/datasets/arrow_dataset.py

```diff
@@ -4140,6 +4146,37 @@ def push_to_hub(
         info_to_dump.splits = {
             split: SplitInfo(split, num_bytes=dataset_nbytes, num_examples=len(self), dataset_name=dataset_name)
         }
+        if config.DATASETDICT_INFOS_FILENAME in repo_files:
+            download_config = DownloadConfig()
+            download_config.download_desc = "Downloading metadata"
+            download_config.use_auth_token = token if token is not None else HfFolder.get_token()
+            dataset_infos_path = cached_path(
+                hf_hub_url(repo_id, config.DATASETDICT_INFOS_FILENAME),
+                download_config=download_config,
+            )
+            with open(dataset_infos_path, encoding="utf-8") as f:
+                dataset_infos: DatasetInfosDict = json.load(f)
+                repo_info = DatasetInfo.from_dict(dataset_infos[next(iter(dataset_infos))])
+            logger.warning("Updating downloaded metadata with the new split.")
+
+            if len(repo_info.splits) > 1 or (len(repo_info.splits) == 1 and next(iter(repo_info.splits)) != split):
```

Comment:
maybe simpler this way
```suggestion
            if not repo_info.splits or list(repo_info.splits) != [split]:
```